pkg/datapath/linux/ipsec/ipsec_linux_test.go

```diff
@@ -69,6 +69,61 @@ func (p *IPSecSuitePrivileged) TestLoadKeys(c *C) {
 	c.Assert(err, IsNil)
 }
 
+func (p *IPSecSuitePrivileged) TestDeleteXfrm(c *C) {
+	_, local, err := net.ParseCIDR("1.2.3.1/24")
+	c.Assert(err, IsNil)
+	_, remote, err := net.ParseCIDR("1.2.3.2/24")
+	c.Assert(err, IsNil)
+
+	// Build a key so we can add a state entry
+	_, authKey, err := decodeIPSecKey("0123456789abcdef0123456789abcdef")
+	c.Assert(err, IsNil)
+	_, cryptKey, err := decodeIPSecKey("0123456789abcdef0123456789abcdef")
+	c.Assert(err, IsNil)
+	key := &ipSecKey{
+		Spi:   1,
+		ReqID: 1,
+		Auth:  &netlink.XfrmStateAlgo{Name: "hmac(sha256)", Key: authKey},
+		Crypt: &netlink.XfrmStateAlgo{Name: "cbc(aes)", Key: cryptKey},
+	}
+
+	// Test adding state and policy and removing it
+	ipSecKeysGlobal[""] = key
+
+	xfrmStateList, err := netlink.XfrmStateList(0)
+	c.Assert(err, IsNil)
+	xfrmPolicyList, err := netlink.XfrmPolicyList(0)
+	c.Assert(err, IsNil)
+
+	stateLen := len(xfrmStateList)
+	policyLen := len(xfrmPolicyList)
+
+	_, err = ipSecReplaceState(local.IP, remote.IP)
+	c.Assert(err, IsNil)
+
+	err = ipSecReplacePolicyOut(local, remote)
+	c.Assert(err, IsNil)
+
+	xfrmStateList, err = netlink.XfrmStateList(0)
+	c.Assert(err, IsNil)
+	xfrmPolicyList, err = netlink.XfrmPolicyList(0)
+	c.Assert(err, IsNil)
+
+	c.Assert(stateLen+1, Equals, len(xfrmStateList))
```

Comment:
Why does replacing state result in a state list that's longer than the prior list?